src-unix/FileIO.hs

```diff
@@ -16,7 +19,11 @@ data FHandle = FHandle Fd
 
 -- should handle opening flags correctly
 open :: FilePath -> IO FHandle
+#if !MIN_VERSION_unix(2,8,0)
 open filename = fmap FHandle $ openFd filename WriteOnly (Just stdFileMode) defaultFileFlags
+#else
+open filename = fmap FHandle $ openFd filename WriteOnly defaultFileFlags
```

Comment:
Likely wrong, see [takano-akio/filelock@`d93257d` (#13)](https://github.com/takano-akio/filelock/pull/13/commits/d93257d1721e4e711b64a0448f417e1fdb080c71).  Should be:
```haskell
open filename = fmap FHandle $ openFd filename WriteOnly defaultFileFlags{ creat = Just stdFileMode }
```